Firebase/Messaging/FIRMessaging.m

```diff
@@ -233,28 +240,19 @@ + (void)load {
                                  withInstanceID:[FIRInstanceID instanceID]
                                withUserDefaults:[GULUserDefaults standardUserDefaults]];
     [messaging start];
+    [messaging configureNotificationSwizzlingIfEnabled];
     return messaging;
   };
   FIRComponent *messagingProvider =
       [FIRComponent componentWithProtocol:@protocol(FIRMessagingInstanceProvider)
-                      instantiationTiming:FIRInstantiationTimingLazy
+                      instantiationTiming:FIRInstantiationTimingEagerInDefaultApp
                              dependencies:@[ analyticsDep ]
                            creationBlock:creationBlock];
 
   return @[ messagingProvider ];
 }
 
-+ (void)configureWithApp:(FIRApp *)app {
```

Comment:
do we need to fix the same for other SDK as well, such as InstanceID, remote config etc?

Reply:
Apologies for the delay - IID was fixed in #4030, will be addressing the other SDKs in a separate PR.